lib/plugins/manifest.js

```diff
@@ -24,11 +24,14 @@ module.exports = function(plugins, webpackConfig) {
         manifestPrefix = webpackConfig.publicPath.replace(/^\//, '');
     }
 
-    plugins.push(new ManifestPlugin({
+    let defaultConfig = {
         basePath: manifestPrefix,
         // guarantee the value uses the public path (or CDN public path)
         publicPath: webpackConfig.getRealPublicPath(),
         // always write a manifest.json file, even with webpack-dev-server
         writeToFileEmit: true,
-    }));
+    };
+    let manifestConfig = Object.assign(defaultConfig, webpackConfig.manifestPluginOptions);
```

Comment:
Just a minor remark here, `Object.assign` modifies its first argument, so in this case `defaultConfig` (which doesn't really makes sense since it won't be the default config anymore after that).

You probably could either do:

```js
const defaultConfig = { /* ... */ };
const manifestConfig = Object.assign({}, defaultConfig, webpackConfig.manifestPluginOptions);
```

Or only use a single variable (which looks a bit less clear in my opinion):

```js
const manifestConfig = Object.assign({}, { /* ... */ }, webpackConfig.manifestPluginOptions);